src/sync/sync.go

```diff
@@ -0,0 +1,125 @@
+package sync
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	// "github.com/codegangsta/negroni"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/gorilla/mux"
+)
+
+var applications map[string][]Connection
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+	CheckOrigin: func(r *http.Request) bool {
+		return true
+	},
+}
+
+type SyncServer struct {
+	r *mux.Router
+}
+
+// ServeHTTP serves requests from the EaseServer's mux while allowing
+// cross origin access.
+func (s *SyncServer) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	if origin := req.Header.Get("Origin"); origin != "" {
+		rw.Header().Set("Access-Control-Allow-Origin", origin)
+		rw.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
+		rw.Header().Set("Access-Control-Allow-Headers",
+			"Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")
+	}
+	// Stop here if its Preflighted OPTIONS request
+	if req.Method == "OPTIONS" {
+		return
+	}
+
+	// Lets Gorilla work
+	s.r.ServeHTTP(rw, req)
+}
+
+func createRouting() *mux.Router {
+	router := mux.NewRouter()
+	router.HandleFunc("/sub", subHandler)
+	router.HandleFunc("/pub/{application}", pubHandler)
```

Comment:
Here `{application}` refers to the application name, correct?  I would change it to reflect that, `app_name` or `application_name`.  Additionally, because of the way we're storing applications, you'll need the username here as part of the path as well.